### ISSUE TITLE
Need information for SilentCryptoMiner

### ISSUE THREAD

**Issue by Kusursuz31** [UnamSanctam/UnamWebPanel#374]
Hello, i am trying to use binance pool but i couldn't. Is there any example for binance pool?

Thats binance pool example:
![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/1ff3e195-e2c3-4c09-9ca0-4d33ad637ed4)

I used with nbminer and it's worked. Usage for RVN on NbMiner "-a kawpow -o stratum+tcp://rvn.poolbinance.com:9000 -u seytanpool.{randomid} (yourwallet here) -p 123456", i really need to use binance on SilentCryptoMiner.

I hope you'll reply and solve my problem.

Other question:
if Compatible GPU found: False is miner continue to work or it will stop?

**Comment by UnamSanctam** [comment]
It would be something like this if following the configuration you posted.
Pool:
`rvn.poolbinance.com:9000`
Wallet:
`saytanpool.{COMPUTERNAME}`
(you can use something else instead of {COMPUTERNAME}, not really sure what it wants there but {COMPUTERNAME} will replace that with the computers name)
Password:
`123456`

>if Compatible GPU found: False is miner continue to work or it will stop?

The message means that your computer does not have any (dedicated) GPU and thus can't mine using a GPU.

**Comment by Kusursuz31** [comment]
I tested like that but it's not showed up in binance have to test again.

How can I find out crypto's system requirements? Like VRAM or Supported GPUS.

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/243ce90d-2e5d-4c79-8af1-30f7dc8d7e8e)

Thats my test PC. Compatible GPU is TRUE but it doesn't showed up any RVN or ETC mining result. Just showing monero.

Do you have any suggestion for dig any good coin?

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/d006784f-055c-4536-ac94-dbc79b1a42f9)

Other questions for WebPanel:

I'm installed WebPanel for preivew with xampp, but i am confused to import database. It's not SQL so i can't use phpmyadmin.

How can i install database file?

**Comment by UnamSanctam** [comment]
The GPU is an AMD Radeon HD 6570, which has only 2GB of VRAM, so neither RVN or ETC can be mined due to them requiring more VRAM than that, you can find some DAG sizes (how much VRAM is needed to mine that coin) here: https://minerstat.com/dag-size-calculator and GPU coin probabilities here: https://whattomine.com/

>Other questions for WebPanel:
I'm installed WebPanel for preivew with xampp, but i am confused to import database. It's not SQL so i can't use phpmyadmin.
How can i install database file?

It uses SQLite as a database, so there's no need to install anything, it should work just like it is right now.

**Comment by Kusursuz31** [comment]
Alr, thanks for the informations.

So, i don't need to import any SQL right? It will auto add zombies in panel?

And last question: How can i use this panel with miner?

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/38b6e637-c816-4f62-9111-c3c136c58a8b)

I mean where i need to put what url to access miner from panel?

**Comment by UnamSanctam** [comment]
Yes there's no need to import any SQL. To use the web panel with the miner you should enable  "API Endpoint URL" and put the URL into the box below it, like `http://DomainOrIP/api/endpoint.php`

It's also recommended to use the "Remote Configuration" option, so that you can change settings whenever you need. You should enter a RAW URL from somewhere like Pastebin or rentry into it and it will read any settings from it and use those (the settings should be in a JSON format, like how it looks if you press the "JSON" tab in your screenshot.)

**Comment by Kusursuz31** [comment]
"Remote Configuration" is can't remoteable in panel? I thought that's what it was. Or it for to take precautions?

**Comment by UnamSanctam** [comment]
It can be changed in the panel, but if something ever happens to your panel then you can still change settings in the "Remote Configuration".

**Comment by Kusursuz31** [comment]
So, yes it for to take precautions.

But i wondering how can i change all zombies config one by one with "Remote Configuration"? Cuz it will not work one by one. It will change for all zombies right?

**Comment by UnamSanctam** [comment]
The "Remote Configuration" changes all of them, but you can also change the web panel URL with the "Remote Configuration", so if anything ever happens to your web panel (like a new domain or IP) then you can change the web panel URL for all your miners using the "Remote COnfiguration".

**Comment by Kusursuz31** [comment]
Ohh that really awesome feature. So i can use for only change my domain. Thats really helpful.

Can you give example json for pastebin or anywhere with Endpoint configuration?

I mean for Endpoint configuration like this:

{
    "algo": "rx/0",
    "pool": "xmr-eu1.nanopool.org",
    "port": 10343,
    "wallet": "wallet.{COMPUTERNAME}/WithdrawalPassword",
    "password": "",
    "nicehash": false,
    "ssltls": true,
    "max-cpu": 20,
    "idle-wait": 5,
    "idle-cpu": 80,
    "stealth-targets": "Taskmgr.exe,ProcessHacker.exe,perfmon.exe,procexp.exe,procexp64.exe",
    "kill-targets": "",
    "stealth-fullscreen": false,
    "endpoint-configuration":"https:///api/endpoint.php"
}

How can i add this "endpoint-configuration" part to add pastebin or anywhere? I hope you understand me.

**Comment by UnamSanctam** [comment]
Yes exactly, it would be like that, the option is called "api-endpoint":

````
{
"algo": "rx/0",
"pool": "xmr-eu1.nanopool.org",
"port": 10343,
"wallet": "wallet.{COMPUTERNAME}/WithdrawalPassword",
"password": "",
"nicehash": false,
"ssltls": true,
"max-cpu": 20,
"idle-wait": 5,
"idle-cpu": 80,
"stealth-targets": "Taskmgr.exe,ProcessHacker.exe,perfmon.exe,procexp.exe,procexp64.exe",
"kill-targets": "",
"stealth-fullscreen": false,
"api-endpoint":"https://DomainOrIP/api/endpoint.php"
}
````
Which you should upload as an unlisted paste on something like Pastebin, and then use the "raw" link in the miner (old example: https://pastebin.com/raw/mpRWix6f)

**Comment by Kusursuz31** [comment]
Thank you very much for everything, I will follow your suggestions. You're the best!

**Comment by UnamSanctam** [comment]
No problem, good luck with your mining!

**Comment by Kusursuz31** [comment]
BTW, I will try to translate your panel into Turkish. If I can translate it successfully, I will send it to you to add to the source.

**Comment by Kusursuz31** [comment]
[UnamWebPaneL.zip](https://github.com/user-attachments/files/16181755/UnamWebPaneL.zip)

There'u go We translated too many words.

You can give credit only for my friend https://github.com/KnullXD


Btw we found a bug:

When you press copy button it showing pop-up but everything is white.

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/70ab4afc-29b0-4452-92e0-48bcf125c7cb)

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/70fc7ee0-82e8-4e26-9374-e80755ea4e8a)

**Comment by Kusursuz31** [comment]
I tried binance pool but my miner is didn't get any IP address and get Error.

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/8a4ea85d-c11c-419b-9c8d-3ff97c011ab0)

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/4f052da0-2d59-4472-ba91-10c00961c6eb)

**Comment by UnamSanctam** [comment]
I don't believe the binance RVN pool ports support SSL, so you need to change the "Connection Scheme" from `stratums` to `stratum`

**Comment by Kusursuz31** [comment]
What about IP address?

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/9802a33b-00a7-4c30-9136-f272369a5bd3)


Btw i changed pool to 2miners but still getting Error.

**Comment by UnamSanctam** [comment]
::1 means localhost, it's your local IP. Did you change the connection scheme to `stratum` instead of `stratums`? For 2miners RVN port 6060 is the non-SSL port and port 16060 is the SSL port.

**Comment by Kusursuz31** [comment]
![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/7aade864-e69a-409a-90f2-f828f7b96cca)

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/fba3e48c-e49c-491a-b372-26624dbdbd03)


![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/7507d268-e995-4fc9-b39e-b9485d6e6ca0)

Problem is solved but now i have clock problem:

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/db033155-73b8-478f-9d00-5a41262ad472)

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/59027739/59ca9d05-2f95-49d9-968c-bc097df74a83)

**Comment by Kusursuz31** [comment]
Hello, i am currently trying on website.

I installed plesk in my ubuntu server.

Currently i have problem with CloudFlare:

![image](https://github.com/user-attachments/assets/1666e28e-b29d-4e19-9641-f1ccabef2575)

https://domain.com/pages/miners.php I have SSL and using https. "162.158.19.146" thats IP address is not real it's CloudFlare IP:

![image](https://github.com/user-attachments/assets/b5b51ca6-fa68-4fdf-8f87-78b31fc7f039)

Do you know how can i solve that problem?

Other problem:

This is my pc:
![image](https://github.com/user-attachments/assets/15ecfd3e-e7ae-4275-b911-a7bdd70af99b)

Client result:
![image](https://github.com/user-attachments/assets/379f458a-a59b-4d5a-9f05-16b4cdece5e7)

Panel result:

![image](https://github.com/user-attachments/assets/82540d51-61a3-40af-9f97-edd295a19ec8)


I'm been waiting 20 min. Still didn't come in panel.

**Comment by UnamSanctam** [comment]
The clock in your web server being different from your computer can have many reasons, though the easiest way to change it is to manually specify the timezone inside the web panel code, though I don't know how that will interact with the database.

For your Cloudflare, do you mean that it shows the Cloudflare IP and not the miners? Then that is because all the traffic is actually coming from Cloudflare so the only IP the web server can know is Cloudflares. You can technically retrieve the original IP but it requires changes to your web server and then in the code of the web panel https://developers.cloudflare.com/support/troubleshooting/restoring-visitor-ips/restoring-original-visitor-ips/.  

For the miner that does not appear, if you enter the URL that's inside the `--cinit-api` in a browser on the same computer that is mining does the website work properly?

**Comment by Kusursuz31** [comment]
I changed my timezone in plesk I'll check later.

Yes, it shows the Cloudflare IP and not the miners.

"Then that is because all the traffic is actually coming from Cloudflare so the only IP the web server can know is Cloudflares. You can technically retrieve the original IP but it requires changes to your web server and then in the code of the web panel" I read your sent page but i didn't understand.

How can i check with --cinit-api? i mean https://domain.com/api/endpoint.php--cinit-api like that or what?

Edit:

when i put this url on mining pc https://domain.com/api/endpoint.php it says:

Endpoint is up and running. This page is shown since no data was posted during the request or the data posted was invalid.

**Comment by UnamSanctam** [comment]
>Yes, it shows the Cloudflare IP and not the miners.
"Then that is because all the traffic is actually coming from Cloudflare so the only IP the web server can know is Cloudflares. You can technically retrieve the original IP but it requires changes to your web server and then in the code of the web panel" I read your sent page but i didn't understand.

The web panel has no idea what the original computers IP was before it was sent through Cloudflare, so the only way to find out the original IP is from what Cloudflare tells you. They do it through a special HTTP header but it requires some setup to do depending on what kind of web server you're running.

>when i put this url on mining pc https://domain.com/api/endpoint.php it says:
Endpoint is up and running. This page is shown since no data was posted during the request or the data posted was invalid.

That means that it's working, and it says that on the computer where the miner doesn't appear in the web panel?

**Comment by Kusursuz31** [comment]
I'm currently changed a domain using only http without cloudflare.

I tested on localhost and it showing up in 2 mins. But it's not showing up in my webpanel.

It's using http.

And yes "Endpoint is up and running. This page is shown since no data was posted during the request or the data posted was invalid." it saying this on my pc. I'm opening miner on my pc to see results.

**Comment by Kusursuz31** [comment]
My PHP Version: 8.3.9
API Endpoint: http://mydomain.com/api/endpoint.php
Hosting: VPS / ubuntu

Endpoint webpanel is working on localhost:

![image](https://github.com/user-attachments/assets/7e4d20b2-55b6-4589-a884-a95578ee2b61)

Endpoint result: Endpoint is up and running. This page is shown since no data was posted during the request or the data posted was invalid.

Working on localhost but it's not working on web panel.

**Comment by UnamSanctam** [comment]
And some work on your Cloudflare panel but not others? Or do none work? Is the Cloudflare panel the same server as your localhost one except just with a tunnel?

**Comment by Kusursuz31** [comment]
Currently I'm not using CloudFlare. I have only hosting dns.

**Comment by UnamSanctam** [comment]
Can you give some more information on how you've set everything up? Do you have any miners at all in your other web panel (not the localhost one)?

**Comment by Kusursuz31** [comment]
No, i am not started but i am tested with my machine. Miners are not showing up on web panel but they showing up on pool.

My setup steps:

I connected my domain on my hosting.

I put my files on here:
![image](https://github.com/user-attachments/assets/a1e5eb7f-289d-4b32-93cf-d9d262c8a58e)

Everything is fine for setup.
![image](https://github.com/user-attachments/assets/993e4bc5-7a96-4436-b2a2-42edfb032211)

I can create configuration:
![image](https://github.com/user-attachments/assets/884fc9b1-f9d1-49e3-b08a-ae45cb6aaf5f)

Now everything is done for webpanel. Let's build miner.

![image](https://github.com/user-attachments/assets/02a292ac-5fac-4c2b-ab22-e5928115b7a0)

I'm using http cuz, i don't have SSL.

![image](https://github.com/user-attachments/assets/5f62f893-2055-4efd-bf2b-828cb4804c28)

Build succesfuly. Lets test on machine.
![image](https://github.com/user-attachments/assets/d33d10c0-06ca-4d86-9484-44e676ca68a7)

I am closed windows defender:

![image](https://github.com/user-attachments/assets/4217a244-544c-4179-a9ee-70461f554769)

Started miner with admin after started checker:

![image](https://github.com/user-attachments/assets/5aaaa28f-c1a4-4b37-96d9-cee5b514781e)

My endpoint result:
![image](https://github.com/user-attachments/assets/33672d61-4006-421a-a0b8-a04c4dcec275)

Let's check web panel after 2 or 3 min.

![image](https://github.com/user-attachments/assets/e24c14f7-5035-42b2-bf96-eb794f88b6d6)

No miner's here. But it's showing up in pool.

![image](https://github.com/user-attachments/assets/ea85f7dd-9c84-4b9d-b403-df6895b2b019)

**Comment by Kusursuz31** [comment]
I solved myself.

I installed SSL and used https in miner build and result:

![image](https://github.com/user-attachments/assets/90947dd1-46e8-4979-b51a-a1af97cff601)

**Comment by UnamSanctam** [comment]
Alright, then it's possible that your host might have been blocking the non-SSL traffic (since it could see the data being sent).

**Comment by Kusursuz31** [comment]
Update:

![image](https://github.com/user-attachments/assets/21502326-4dc6-4ef7-815e-8b463fdcb32f)

We baited 2 people they have stupid GPU's but they not came in panel only came 1 people.

![image](https://github.com/user-attachments/assets/5d47bfa0-bc64-4665-b331-5d21556dd103)

Is it normal or what?

**Comment by UnamSanctam** [comment]
Only one of them is an actual GPU, the Intel one is just the integrated graphics of the CPU, so it won't mine on that (because it can't really, and even if it could it would not be desirable).

**Comment by Kusursuz31** [comment]
Yes i know but i wondering if they have good cpu maybe i can use but they not come.

BTW he's gone:
![image](https://github.com/user-attachments/assets/621ca46b-6d2c-4ddc-a364-f2b7266c97f9)

I guess it's Defender Exclusions problem.

**Comment by UnamSanctam** [comment]
You don't want to GPU mine on CPU's since it will usually cause a lot of lag (freezing usually, where the computer freezes every few seconds). For the one that's offline, it's hard to know, but are you not installing it as System?

**Comment by Kusursuz31** [comment]
You didn't understand me. I mean some cryptos are supporting cpu mining like XMR and if they have bad GPU but have good CPU i can change the config to mining XMR.

But they're looking offline, i guess they removed the our miner.

I'm extracting my miner on "C:\ProgramData" cuz i saw you're adding whitelist.

But i guess is not working some times or they not closing windows defender.

Screenshot from log:
![image](https://github.com/user-attachments/assets/a9c4f345-ee48-4f61-bb09-e02ed46a3c19)

Other questions:

If I turn on the startup option in miner can it be seen in Task Manager? 

Like this screenshot:
![image](https://github.com/user-attachments/assets/200b5a0a-fbfa-4056-b654-a749b414557e)

And what is stealth on fullscren?

**Comment by UnamSanctam** [comment]
>You didn't understand me. I mean some cryptos are supporting cpu mining like XMR and if they have bad GPU but have good CPU i can change the config to mining XMR.

You can always mine CPU and GPU at the same time, since they don't use the same kinds of resources, so you can add both an XMR miner and RVN miner to the miner.

>But they're looking offline, i guess they removed the our miner.
I'm extracting my miner on "C:\ProgramData" cuz i saw you're adding whitelist.
But i guess is not working some times or they not closing windows defender.

Yes it will add exclusions if you run the miner as administrator. If you open up those detections and check the details what do they say? And can you see the exclusions if you check the exclusions in that Windows Defender?

>Other questions:
If I turn on the startup option in miner can it be seen in Task Manager?
Like this screenshot:

Not if it's installed as an administrator, in which case it will be installed as a Windows Service. But since we can see the "Active Window" and non-system "Username" in the web panel then I assume you're not running it as administrator.

>And what is stealth on fullscren?

The option will cause the miner to pause whenever there's an app open that triggers the "fullscreen mode" for notifications in Windows, it's quite complicated but I can't link the wiki with the explanation since the repository is disabled.

**Comment by Kusursuz31** [comment]
>  You can always mine CPU and GPU at the same time, since they don't use the same kinds of resources, so you can add both an XMR miner and RVN miner to the miner.

So I can use both.

> Yes it will add exclusions if you run the miner as administrator. If you open up those detections and check the details what do they say? And can you see the exclusions if you check the exclusions in that Windows Defender?

![image](https://github.com/user-attachments/assets/3ca7df99-e1ae-42c7-8622-2830345d2ef0)

I can't open detection details cuz, it's zombie pc not mine. But we tested on test pc and it's added proof:
![image](https://github.com/user-attachments/assets/8bc5a2f7-40a8-4d31-a923-f89c4c0ecbfd)

> Not if it's installed as an administrator, in which case it will be installed as a Windows Service. But since we can see the "Active Window" and non-system "Username" in the web panel then I assume you're not running it as administrator.

I'm created a another app to start adminastrator and opened Administrator bool to true in miner build.

I didn't understand "But since we can see the "Active Window" and non-system "Username" in the web panel then I assume you're not running it as administrator." This sentence.

>The option will cause the miner to pause whenever there's an app open that triggers the "fullscreen mode" for notifications in Windows, it's quite complicated but I can't link the wiki with the explanation since the repository is disabled.

Alr, i don't want any fullscreen method.

**Comment by UnamSanctam** [comment]
>So I can use both.

Yes.

>I can't open detection details cuz, it's zombie pc not mine. But we tested on test pc and it's added proof:

Alright good.

>I'm created a another app to start adminastrator and opened Administrator bool to true in miner build.
I didn't understand "But since we can see the "Active Window" and non-system "Username" in the web panel then I assume you're not running it as administrator." This sentence.

If the miner is installed as administrator (as a Windows Service) then the "Active Window" field should say "Running as System" and the "Username" should say "System". What does it say if you check your "Startup" tab in the miner build?

**Comment by Kusursuz31** [comment]
>If the miner is installed as administrator (as a Windows Service) then the "Active Window" field should say "Running as System" and the "Username" should say "System". What does it say if you check your "Startup" tab in the miner build?

Do you have any screenshot on panel? Cuz my Active Window is working on only applications.

![image](https://github.com/user-attachments/assets/96a50699-470a-479c-bafd-6516cac44315)

I didn't enabled this feature:

![image](https://github.com/user-attachments/assets/f19a3df7-e1d0-4c6f-8559-7d26dd84f6f6)

**Comment by UnamSanctam** [comment]
Alright, then it says that in "Active Window" and "Username" because you're not installing the miner, if "Startup" (installation) isn't enabled then it won't install the miner (and thus won't start when the computer restart), instead it will just run it once.

**Comment by Kusursuz31** [comment]
My project is starting up miner when windows start. But I'll enable startup in build.

**Comment by Kusursuz31** [comment]
I have another problem on web panel.

User is giving my wallet and it's have too many characters it's broking panel.
![image](https://github.com/user-attachments/assets/34d78b9e-aab6-4e3a-a031-5302bfd9a6ff)

Example not broked:

![image](https://github.com/user-attachments/assets/94067595-5644-49be-82b0-fe53bdd1be0e)

Do you have any idea to solve that problem?

**Comment by UnamSanctam** [comment]
That's how it works, you can press the '+' at the beginning to see al the information of a miner.

**Comment by Kusursuz31** [comment]
Yeah i know i can see with press '+' but on mobile it's not responsive and getting some problem.

**Comment by UnamSanctam** [comment]
It should work on mobile just fine (the plus thing is part of the responsive option of the datatables plugin), only way to remove the wallet/user would be to hide or remove it inside the code of your web panel.

**Comment by Kusursuz31** [comment]
Should be work yea lemme try.

**Comment by Kusursuz31** [comment]
Nevermind, i have another question.

![image](https://github.com/user-attachments/assets/828dd4c4-e832-4252-abb6-b03a12ffa2b4)

what is different in mining and mining(idle)?

**Comment by UnamSanctam** [comment]
During normal mining it will use the "Max CPU" or "Max GPU" setting and during idle mining it will use the "Idle CPU" or "Idle GPU" setting.

**Comment by Kusursuz31** [comment]
Alr, thanks! btw i changed administrator access now it's showing up SYSTEM on user:

New build:
![image](https://github.com/user-attachments/assets/762f8c0f-7ed0-42bf-b476-6727b4d77c3b)

Old build:
![image](https://github.com/user-attachments/assets/318f96da-7932-418e-b024-cb4eadf1c668)

**Comment by UnamSanctam** [comment]
Alright great, then it means it's installed, also means that the watchdog should be running (unless you disabled that).

**Comment by Kusursuz31** [comment]
I haven't disabled watchdog but I wonder what it does?

**Comment by UnamSanctam** [comment]
When the miner get injected the program also injects another process called the watchdog, the watchdog constantly checks whether the miners are running or not, if they aren't running then it starts them. The watchdog also constantly checks whether the installed Windows Service or program file (the .exe that gets installed) have been removed, if they have been removed then it places back the .exe and installs the service again (and also re-adds the Defender exclusions before doing that).

**Comment by Kusursuz31** [comment]
Wow thats really good feature. Thanks for the reply. I will keep you updated.

**Comment by Kusursuz31** [comment]
Hey, now I'll ask questions and I will share my results.

Currently we have problem with admin panel.

1-) If someone open miner It's showing 2 users with same id. (1 CPU and 1 GPU) Like:
![image](https://github.com/user-attachments/assets/2d50e53f-e154-4b72-b4f8-8e01c0da6e9a)

2-) It says miner is running but it's say 0H/s:
![image](https://github.com/user-attachments/assets/94e32c71-4cd5-4868-9ef6-977f6e8f791f)

3-) I'm currently using binance pool but hashrate's are different on panel and pool.

Binance:
15M: 28.19 MH/s 24H: 20.43 MH/s
![image](https://github.com/user-attachments/assets/dc58b4d4-0193-49be-989d-b10151789731)

Panel:
Total: 25.1‬ MH/s
![image](https://github.com/user-attachments/assets/c2699227-a46c-4975-b1ea-9e7e49db17b2)

4-) Panel need to update. My Ideas:

Need to add filter to list users like Show CPU or GPU users.

In Statistics page need to show total MH/s value and online CPU or GPU users.

And someone is opening my miner and showing up in panel but they are going offline sometimes. I have 200 users but i only saw max 12 users for gpu with cpu max 30 users.

**Comment by UnamSanctam** [comment]
>1-) If someone open miner It's showing 2 users with same id. (1 CPU and 1 GPU) Like:

Because one is the GPU miner and one is the CPU miner, the web panel shows the miners, not the users. The miner entries have different information.

>2-) It says miner is running but it's say 0H/s:

Likely because it's a Nvidia GT 730, which does not have enough VRAM to mine RVN.

>3-) I'm currently using binance pool but hashrate's are different on panel and pool.

That's normal, the web panel shows the real hashrate of the miners, the pool only shows a loosely calculated hashrate based upon the amount of shares submitted during some interval.

>4-) Panel need to update. My Ideas:

>Need to add filter to list users like Show CPU or GPU users.

What would it be used for? And if you need to filter miners you might be able to do so by entering "CPU Miner" or "GPU Miner" into the search field. Or you might need to enter "xmrig" or "ethminer"

>In Statistics page need to show total MH/s value

The total hashrate for each algorithm (have to be differentiated because hashrate is an arbitrary number that changes depending on the algorithm) can be seen if `hashrate_history` is enabled for miners.

>and online CPU or GPU users.

It does currently show the amount of computers/users that are online, though I can add specific GPU/CPU miners sections if you want. They just aren't there at the moment because they would take up a lot of space.

>And someone is opening my miner and showing up in panel but they are going offline sometimes. I have 200 users but i only saw max 12 users for gpu with cpu max 30 users.

That's not due to the web panel, do you mean that you actually have 200 people with miners running (no one is offline) but only 12/30 show up in the web panel? Or do you just mean that you have the miner installed on 200 people? Since people can turn their computer off, and in normal cases most computers will be turned off when not being used.

**Comment by Kusursuz31** [comment]
>Because one is the GPU miner and one is the CPU miner, the web panel shows the miners, not the users. The miner entries have different information.

It's okay.

>Likely because it's a Nvidia GT 730, which does not have enough VRAM to mine RVN.

It's doing sometimes with other GPU's like 1650 super 1660 and 1050 TI.

![image](https://github.com/user-attachments/assets/11d49d9e-6b41-4020-87e5-5fc932f02449)

>That's normal, the web panel shows the real hashrate of the miners, the pool only shows a loosely calculated hashrate based upon the amount of shares submitted during some interval.

That's what I thought.

>What would it be used for? And if you need to filter miners you might be able to do so by entering "CPU Miner" or "GPU Miner" into the search field. Or you might need to enter "xmrig" or "ethminer"

It's for Statistics page. I checking my all users manually to see my total hashrate for CPU and GPU. If can possible can you add card to see my active CPU mining, GPU mining and their hashrates?

>It does currently show the amount of computers/users that are online, though I can add specific GPU/CPU miners sections if you want. They just aren't there at the moment because they would take up a lot of space.

It's really okay I'll create example with paint to show what i want.

![image](https://github.com/user-attachments/assets/48b7ef57-d40c-4295-9042-24547dfef5e2)

>That's not due to the web panel, do you mean that you actually have 200 people with miners running (no one is offline) but only 12/30 show up in the web panel? Or do you just mean that you have the miner installed on 200 people? Since people can turn their computer off, and in normal cases most computers will be turned off when not being used.

Currently we have 213 users. But someone is showing up panel only a few mins and they're going to offline after a hours. I checked they all days but they're still offline.

![image](https://github.com/user-attachments/assets/1e2183fa-5483-4544-8581-b9048bfeb220)

I think they are removed our miner.

**Comment by Kusursuz31** [comment]
Yo unam i guess you forget to answer. :D @UnamSanctam

**Comment by UnamSanctam** [comment]
Yeah sorry, someone has been spamming my email constantly for some reason so I'm missing notifications. 

>It's doing sometimes with other GPU's like 1650 super 1660 and 1050 TI.

Those only have 4GB of VRAM, so they don't have enough to mine RVN either.

>It's for Statistics page. I checking my all users manually to see my total hashrate for CPU and GPU. If can possible can you add card to see my active CPU mining, GPU mining and their hashrates?
>It's really okay I'll create example with paint to show what i want.

You can enable `hashrate_history` in the config.php if you want to see the current and historical hashrates for every algorithm. The hashrates have to be divided into algorithms since each algorithm uses arbitrary hashrate numbers.  Though enabling hashrate history will cause more space to be used, and make the endpoint a bit slower (since it has to write and clean up the history). I can also add boxes just for the current hashrate of each algorithm (with no history).

>Currently we have 213 users. But someone is showing up panel only a few mins and they're going to offline after a hours. I checked they all days but they're still offline.
>I think they are removed our miner.

Yes then I assume the miner was removed, likely after the miner had been injected (and thus been added into the web panel).

**Comment by Kusursuz31** [comment]
>Those only have 4GB of VRAM, so they don't have enough to mine RVN either.
Someone have Min 4GB VRAM but they still not mining.


>You can enable hashrate_history in the config.php if you want to see the current and historical hashrates for every algorithm. The hashrates have to be divided into algorithms since each algorithm uses arbitrary hashrate numbers. Though enabling hashrate history will cause more space to be used, and make the endpoint a bit slower (since it has to write and clean up the history). I can also add boxes just for the current hashrate of each algorithm (with no history).

I just want to see the total hashrate for gpu and cpu. Like the example picture I posted. Will it show like what I need?

Cuz, it's already true. I gues you didn't understand my idea.

I need to create a card on Statistics page. Like GPU Hashrate and CPU Hashrate. It will show total hashrate while miners are active.

>Yes then I assume the miner was removed, likely after the miner had been injected (and thus been added into the web panel).
I saw sometimes it's asking administrator access to add whitelist. Maybe someone is not giving that access and defender is catching our miner.

**Comment by Kusursuz31** [comment]
@UnamSanctam Yo, whassup?

**Comment by UnamSanctam** [comment]
Yes hashrate history would show that, you can't show GPU hashrate and CPU hashrate since if someone mines different algorithms then that would break instantly since different algorithms have different hashrates on the same GPU. So the hashrate history shows the total hashrate per algorithm, for example your KawPow total hashrate.

**Comment by Kusursuz31** [comment]
Hello @UnamSanctam I have the problem with GPU mining.

![image](https://github.com/user-attachments/assets/48a9c7ad-1bd6-4823-8d27-fdfbe870f4fa)

Some users have 2 GPU Card and it's show 0 H/s Hashrate. Is there any solution?

And It's say stopped for someone:

![image](https://github.com/user-attachments/assets/2cba333b-0169-4c3a-8629-d360f1a0e8ad)

**Comment by UnamSanctam** [comment]
The RTX 4060 Laptop GPU might currently be "disabled" since some laptops do that when no "dedicated program" is running (for example a program that's in the Nvidia Controll Panel list), or there could be something else. It's not uncommon for there to be issues on laptops.

For the MX130 then that can't mine ETC due to only having 2GB of VRAM, and since it says stopped then it's likely that the GPU itself could be missing drivers or other things like that.
